bin/oneoff/back_populate_user_proficiencies_again

```diff
@@ -0,0 +1,357 @@
+#     `conditionals_d2_count` decimal(23,0) DEFAULT NULL,
+#     `conditionals_d3_count` decimal(23,0) DEFAULT NULL,
+#     `conditionals_d4_count` decimal(23,0) DEFAULT NULL,
+#     `conditionals_d5_count` decimal(23,0) DEFAULT NULL,
+#     `max_updated_at` datetime DEFAULT NULL
+#   ) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
+#
+# exported via the mysqldump command
+#
+#   mysqldump dashboard_production user_proficiencies_2016_01
+#     --host=rds-reporting.ceidnfhvpbck.us-east-1.rds.amazonaws.com
+#     --user=db
+#     --password
+#     --result-file=user_proficiencies_2016_01
+#
+# and imported by this script.
+
+require 'csv'
+require_relative '../../dashboard/config/environment'
+
+SUFFIXES = %w(
+  _2016_01
+  _2016_02
+  _2016_03
+  _2016_04
+  _2016_05
+  _2016_06
+  _2016_07
+  _2016_08_partial
+).freeze
+
+# @param line [Array] an array giving UserProficiency values
+# @return [Boolean] a boolean indicating whether basic proficiency has been
+#   achieved.
+def proficient?(line)
+  concept_count = 0
+
+  # Sequencing.
+  if line[3] + line[4] + line[5] >= 3
+    concept_count += 1
+  end
+
+  # Loops.
+  if line[13] + line[14] + line[15] >= 3 ||
+    line[18] + line[19] + line[20] >= 3 ||
+    line[23] + line[24] + line[25] >= 3
+    concept_count += 1
+  end
+
+  # Events.
+  if line[28] + line[29] + line[30] >= 3
+    concept_count += 1
+  end
+
+  # Variables.
+  if line[33] + line[34] + line[35] >= 3
+    concept_count += 1
+  end
+
+  # Functions.
+  if line[38] + line[39] + line[40] >= 3 ||
+    line[43] + line[44] + line[45] >= 3
+    concept_count += 1
+  end
+
+  # Conditionals.
+  if line[48] + line[49] + line[50] >= 3
+    concept_count += 1
+  end
+
+  concept_count >= 3
+end
+
+# @param historical_line [Array] an array giving historical UserProficiency
+#   values
+# @param incremental_line [Array] an array giving incremental UserProficiency
+#   values
+# @return [Array] an array representing the sum of historical_line and
+#   incremental_line
+def increment_user_proficiency(historical_line, incremental_line)
+  sum = historical_line.clone
+
+  (1..50).each do |index|
+    sum[index] += incremental_line[index]
+  end
+
+  unless incremental_line[51].nil?
+    sum[51] = incremental_line[51]
+  end
+
+  if sum[52] && sum.proficient?
+    sum[52] = incremental_line[51]
+  end
+
+  sum
+end
+
+# A to-be-populated hash from user_ids to user_proficiency data.
+user_proficiency_hash = {}
+
+# Process the CSV files, populating user_proficiency_hash from their data.
+SUFFIXES.each do |suffix|
+  filename = "/tmp/user_proficiencies#{suffix}"
+
+  puts "PROCESSING #{filename}..."
+
+  CSV.foreach(filename).with_index do |line, line_num|
+    if line_num % 100_000 == 0
+      puts "  LINE #{line_num}..."
+    end
+
+    # Convert the user_id and level counts from strings to integers.
+    (0..50).each do |index|
+      line[index] = line[index].to_i
+    end
+    # Add a nil value for basic_proficiency_at.
```

Comment:
Index 0: user_id
Index 1-50: level_counts
Index 51: last_progress_at (for the month)
Index 52: basic_proficiency_at